modules/storage-api/src/main/java/org/apache/ignite/internal/storage/UuidIgniteRowId.java

```diff
@@ -0,0 +1,155 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.storage;
+
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+import java.util.UUID;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * UUID-based ignite row id implementation.
+ */
+public final class UuidIgniteRowId implements IgniteRowId {
+    /*
+     * The most significant 64 bits.
+     */
+    private final long mostSigBits;
+
+    /*
+     * The least significant 64 bits.
+     */
+    private final long leastSigBits;
+
+    /**
+     * Constructor.
+     *
+     * @param uuid UUID.
+     */
+    public UuidIgniteRowId(UUID uuid) {
```

Comment:
I think we need to pass a partition here.

Reply:
Ok, not a big deal